src/apps/components/AppAlerts/useSidebarDotState.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useRef, useState } from "react";
+
+import { useSidebarWebhookAlertMetadata } from "./useSidebarWebhookAlertMetadata";
+
+const shouldShowDotCheck = (
+  lastClickDate: string | undefined,
+  lastFailedAttemptDate: string | undefined,
+): boolean => {
+  if (!lastFailedAttemptDate) {
+    return false;
+  }
+
+  return !lastClickDate || lastFailedAttemptDate > lastClickDate;
+};
+
+interface SidebarDotState {
+  handleAppsListItemClick: (clickDate: string) => void;
+  handleFailedAttempt: (failedAttemptDate: string) => void;
+  hasNewFailedAttempts: boolean;
+}
+
+export const useSidebarDotState = (): SidebarDotState => {
+  const { persist, sidebarDotRemoteState } = useSidebarWebhookAlertMetadata();
+
+  const [hasNewFailedAttempts, setHasNewFailedAttempts] = useState(false);
+  const lastClickDateRef = useRef<string | null>(null);
+  const lastFailedAttemptDateRef = useRef<string | null>(null);
+
+  // Initialize state from the remote state
+  useEffect(() => {
+    if (sidebarDotRemoteState) {
+      lastClickDateRef.current = sidebarDotRemoteState.lastClickDate;
+      lastFailedAttemptDateRef.current = sidebarDotRemoteState.lastFailedAttemptDate;
+
+      const shouldShowDot = shouldShowDotCheck(
+        lastClickDateRef.current,
+        lastFailedAttemptDateRef.current,
+      );
+
+      setHasNewFailedAttempts(shouldShowDot);
+    }
+  }, [sidebarDotRemoteState]);
+
+  const handleClick = async (clickDate: string) => {
+    try {
+      await persist({
+        lastClickDate: clickDate,
+        lastFailedAttemptDate: lastFailedAttemptDateRef.current ?? "",
+      });
+    } catch (error) {
+      // Silently ignore errors
```

Comment:
suggestion: I'd think about catch this error and send it to sentry, WDYT?

Reply:
Sure, we can give it a try although I'm afraid we would pollute Sentry with junk errors like 401s